proto/runtime_config_linux.proto

```diff
@@ -0,0 +1,213 @@
+}
+
+// Pids for Linux cgroup 'pids' resource management (Linux 4.3)
+message Pids {
+	// Maximum number of PIDs. A value < 0 implies "no limit".
+	optional int64 Limit = 1;
+}
+
+// BlockIO for Linux cgroup 'blockio' resource management
+message BlockIO {
+	// Specifies per cgroup weight, range is from 10 to 1000
+	optional int64 Weight = 1;
+	// Weight per cgroup per device, can override BlkioWeight
+	optional string WeightDevice = 2;
+	// IO read rate limit per cgroup per device, bytes per second
+	optional string ThrottleReadBpsDevice = 3;
+	// IO write rate limit per cgroup per divice, bytes per second
+	optional string ThrottleWriteBpsDevice = 4;
+	// IO read rate limit per cgroup per device, IO per second
+	optional string ThrottleReadIOpsDevice = 5;
+	// IO write rate limit per cgroup per device, IO per second
+	optional string ThrottleWriteIOpsDevice = 6;
+}
+
+// HugepageLimit structure corresponds to limiting kernel hugepages
+message HugepageLimit {
+	optional string Pagesize = 1;
+	optional int32 Limit = 2;
+}
+
+// Network identification and priority configuration
+message Network {
+	// Set class identifier for container's network packets
+	optional string ClassID = 1;
+	// Set priority of network traffic for container
+	repeated InterfacePriority Priorities = 2;
+}
+
+// InterfacePriority for network interfaces
+message InterfacePriority {
+	// Name is the name of the network interface
+	optional string Name = 1;
+	// Priority for the interface
+	optional int64 Priority = 2;
+}
+
+// Namespace is the configuration for a linux namespace
+message Namespace {
+	// Type is the type of Linux namespace
+	optional string Type = 1;
+	// Path is a path to an existing namespace persisted on disk that can be joined
+	// and is of the same type
+	optional string Path = 2;
+}
+
+// Device represents the information on a Linux special device file
+message Device {
+	// Path to the device.
+	optional string Path = 1;
+	// Device type, block, char, etc.
+  // TODO(vbatts) ensure int32 is fine here, instead of golang's rune
+	optional int32 Type = 2;
+	// Major is the device's major number.
+	optional int64 Major = 3;
+	// Minor is the device's minor number.
+	optional int64 Minor = 4;
+	// Cgroup permissions format, rwm.
+	optional string Permissions = 5;
+	// FileMode permission bits for the device.
+  // TODO(vbatts) os.FileMode is an octal uint32
+	optional uint32 FileMode = 6;
+	// UID of the device.
+	optional uint32 UID = 7;
+	// Gid of the device.
+	optional uint32 GID = 8;
+}
+
+// Seccomp represents syscall restrictions
+message Seccomp {
+  // TODO(vbatts) string instead of "Action" type
+	optional string DefaultAction = 1;
+	repeated Syscall Syscalls = 2;
+}
+
+// Syscall is used to match a syscall in Seccomp
+message Syscall {
+	optional string Name = 1;
+	optional string Action = 2;
+	repeated Arg Args = 3;
+}
+
+// Arg used for matching specific syscall arguments in Seccomp
+message Arg {
+	optional uint32 Index = 1;
```

Comment:
This was a copy from the *.go source. Not entirely clear on this struct. Must be a Seccomp specific syscall mapping. @mrunalp @crosbymichael thoughts?